dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataFlusher.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.upload
+
+import com.datadog.android.core.internal.net.DataUploader
+import com.datadog.android.core.internal.persistence.PayloadDecoration
+import com.datadog.android.core.internal.persistence.file.FileHandler
+import com.datadog.android.core.internal.persistence.file.FileOrchestrator
+
+internal class DataFlusher(
+    internal val fileOrchestrator: FileOrchestrator,
+    internal val decoration: PayloadDecoration,
+    internal val handler: FileHandler
+) : Flusher {
+
+    override fun flush(uploader: DataUploader) {
+        val toUploadFiles = fileOrchestrator.getFlushableFiles()
+        toUploadFiles.forEach {
+            val batch = handler.readData(it, decoration.prefixBytes, decoration.suffixBytes)
+            uploader.upload(batch)
```

Comment:
do we care about `UploadStatus` returned?

Reply:
we do not care here as we are going to clean the files anyway